### PR TITLE
More flexible Makefile help column format

### PR DESCRIPTION
**Pull request by phil-davis**
When formatting the ``make help`` output:
- replace multiple spaces by a single space
- use the ``column`` command to line up the ``Makefile`` targets and descriptions into columns

At the moment we have to get the correct number of spaces in each ``test-xyz:        ## Description of command`` line.

After this change, we can put whatever spaces we like before and after the ``##`` and the ``make help`` still lines up.

### CODE REVIEW

**Comment by phil-davis** [comment]
Inspired by @skshetry doing similar in ransomware_protection.

**Comment by skshetry** [comment]
Would it be too much if I ask for colored target name? 
ref: https://gist.github.com/prwhite/8168133#gistcomment-976470

File: Makefile [hunk 1]
```diff
@@ -100,50 +100,50 @@ clean: clean-deps clean-dist clean-build
 
 ##
```
Comment (skshetry) [inline Makefile]:
```suggestion
##---------------------
```
We need this to separate into multiple sections. Otherwise, it will be very hard to read.

**Comment by phil-davis** [comment]
The colour feature request does not immediately work in my terminal. It will need effort to make sure it will work nicely for all (reasonable) terminals that developers use with *nix and Mac. That can be a weekend PR for "someone who cares".

**Comment by phil-davis** [comment]
Rebased to trigger CI again.

**Pull request merged**